xl-objects.go

```diff
@@ -212,6 +215,25 @@ func getMultipartObjectInfo(storage StorageAPI, bucket, object string) (info Mul
 	return info, nil
 }
 
+// Returns FileInfo - accounts for both single-put and multipart files.
+func (xl xlObjects) statFile(bucket, object string) (FileInfo, error) {
```

Comment:
We shouldn't have functions like this at Object level, we should wrap them as -`getObjectInfo()` and return a proper wrapped result.